### ISSUE TITLE
Qualitative visualization

### ISSUE THREAD

**Issue by vettelx** [3dpcc/GRNet#3]
Hi, very great job! I think your Figure 1 and Figure 7 of the qualitative visualization are very good to show your gains compared with baselines. Would you provide the visualization code or give me some hints? Thanks

**Comment by 3dpcc** [comment]
Hello, Figures 1 and 7 in the paper demonstrate the lossy geometry visualization with RGB information. Specifically, the Ground Truth RGB information is applied to the lossy point cloud geometry through **Average Pooling**. Below is a **code demo** illustrating this:

filename='/media/ivc11506/disk1/lgx/visualization-error_map/GPCC_UP/soldier/gt.ply'
filename_ours='/media/ivc11506/disk1/lgx/visualization-error_map/pretrain_solid_r02/soldier_dgpp_new.ply'

pc,gt_rgb = load_ply_data(filename)
pc_ours = torch.tensor(load_ply_data_geo(filename_ours))

GT_C = ME.utils.batched_coordinates([pc])
ours_C=ME.utils.batched_coordinates([pc_ours])

GT = ME.SparseTensor(features=torch.tensor(gt_rgb).float(),coordinates=GT_C,tensor_stride=[1,1,1])
ours = ME.SparseTensor(features=torch.ones([ours_C.shape[0], 3]).float(),coordinates=ours_C,tensor_stride=[1,1,1])

Avgpool = ME.MinkowskiAvgPooling(kernel_size=5,stride=1,dimension=3)
out_rgb = Avgpool(GT,ours.C)
out_dir = filename_ours+'addrgb.ply'
write_ply_ascii(out_dir,out_rgb.C[:,1:].detach().cpu().numpy(),out_rgb.F.detach().cpu().numpy())

**Comment by vettelx** [comment]
Thanks very much!!!!! Bother to ask about the error map code?

**Comment by 3dpcc** [comment]
[visualization.ipynb.zip](https://github.com/user-attachments/files/16627662/visualization.ipynb.zip)
You can use this.

**Comment by vettelx** [comment]
> [visualization.ipynb.zip](https://github.com/user-attachments/files/16627662/visualization.ipynb.zip) You can use this.

Thank you for your patient and prompt reply!!! Helps me a lot
